src/main/java/io/thinkit/edc/client/connector/services/EdrCache.java

```diff
@@ -0,0 +1,106 @@
+package io.thinkit.edc.client.connector.services;
+
+import static io.thinkit.edc.client.connector.utils.JsonLdUtil.compact;
+import static java.net.http.HttpRequest.BodyPublishers.ofString;
+
+import com.apicatalog.jsonld.JsonLdError;
+import io.thinkit.edc.client.connector.model.*;
+import io.thinkit.edc.client.connector.utils.JsonLdUtil;
+import jakarta.json.JsonArray;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.UnaryOperator;
+
+public class EdrCache {
+    private final String url;
+    private final ManagementApiHttpClient managementApiHttpClient;
+
+    public EdrCache(String url, HttpClient httpClient, UnaryOperator<HttpRequest.Builder> interceptor) {
+        managementApiHttpClient = new ManagementApiHttpClient(httpClient, interceptor);
+        this.url = url;
+    }
+
+    public Result<String> delete(String transferProcessId) {
+        var requestBuilder = HttpRequest.newBuilder()
+                .uri(URI.create("%s/v1/edrs/%s".formatted(this.url, transferProcessId)))
+                .DELETE();
```

Comment:
extracting the request builder in methods will remove some duplication between the sync and the async request (each for every method)